test/dotnet.Tests/CommandTests/Package/Add/GivenDotnetPackageAdd.cs

```diff
@@ -252,6 +290,370 @@ public void VersionRange(bool asArgument)
             cmd.StdErr.Should().BeEmpty();
         }
 
+        [Fact]
+        public void FileBasedApp()
+        {
+            var testInstance = _testAssetsManager.CreateTestDirectory();
+            var file = Path.Join(testInstance.Path, "Program.cs");
+            File.WriteAllText(file, """
+                Console.WriteLine();
+                """);
+
+            new DotnetCommand(Log, "package", "add", "Humanizer@2.14.1", "--file", "Program.cs")
+                .WithWorkingDirectory(testInstance.Path)
+                .Execute()
+                .Should().Pass();
+
+            File.ReadAllText(file).Should().Be("""
+                #:package Humanizer@2.14.1
+
+                Console.WriteLine();
+                """);
+        }
+
+        [Fact]
+        public void FileBasedApp_ReplaceExisting()
```

Comment:
Consider making this a `[Theory]` where the values are `Humanizer` but at different cases. Then use that value inside the original file content. Validating that we do case insensitive comparisons here.